biodatasets/codiesp/codiesp.py

```diff
@@ -0,0 +1,331 @@
+                    "filepath": Path(os.path.join(data_dir, "final_dataset_v4_to_publish/dev")),
+                    "split": "dev",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath, split: str) -> Tuple[int, Dict]:
+        """
+        This method handles input defined in _split_generators to yield (key, example) tuples from the dataset.
+        Method parameters are unpacked from `gen_kwargs` as given in `_split_generators`.
+        """
+
+        paths = {"text_files": Path(os.path.join(filepath, "text_files"))}
+        for task in ["codiesp_d", "codiesp_p", "codiesp_x"]:
+            paths[task] = Path(os.path.join(filepath, f"{split}{task[-1].upper()}.tsv"))
+
+        if self.config.name == "codiesp_D_bigbio_text" or self.config.name == "codiesp_P_bigbio_text":
+            df = pd.read_csv(paths[self.config.subset_id], sep="\t", header=None)
+
+            file_codes_dict = defaultdict(list)
+            for idx, row in df.iterrows():
+                file, code = row[0], row[1]
+                file_codes_dict[file].append(code)
+
+            for guid, (file, codes) in enumerate(file_codes_dict.items()):
+                text_file = Path(os.path.join(paths["text_files"], f"{file}.txt"))
+                example = {
+                    "id": str(guid),
+                    "document_id": file,
+                    "text": text_file.read_text(),
+                    "labels": codes,
+                }
+                yield guid, example
+
+        elif self.config.name == "codiesp_X_bigbio_kb":
+            df = pd.read_csv(paths[self.config.subset_id], sep="\t", header=None)
+
+            task_x_dict = defaultdict(list)
+            for idx, row in df.iterrows():
+                file, label, code, text, spans = row[0], row[1], row[2], row[3], row[4]
+
+                appearances = spans.split(";")
+                spans = []
+                for a in appearances:
+                    spans.append((int(a.split()[0]), int(a.split()[1])))
+
+                task_x_dict[file].append({"label": label, "code": code, "text": text, "spans": spans})
+
+            for guid, (file, data) in enumerate(task_x_dict.items()):
+                example = {
+                    "id": str(guid),
+                    "document_id": file,
+                    "passages": [],
+                    "entities": [],
+                    "events": [],
+                    "coreferences": [],
+                    "relations": []
+                }
+
+                for idx, d in enumerate(data):
+                    example["entities"].append({
+                                            "id": str(guid)+str(idx),
+                                            "type": d["label"],
+                                            "text": [d["text"]],
+                                            "offsets": d["spans"],
+                                            "normalized": [{"db_name": "ICD10-PCS",
```

Comment:
This `"db_name"` should be `ICD10-PCS` if the entity is `PROCEDIMIENTO`, but it should be `ICD10-CM` if the entity type is `DIAGNOSTICO`

Reply:
well seen,  this has been fixed now